client/store/challenge-list.js

```diff
@@ -21,3 +21,14 @@ export const fetchChallenges = () => {
       .then(challenges => dispatch(getChallenges(challenges)));
   };
 };
+
+// Return search challenges with a search term
+export const fetchSearchChallenges = term => dispatch => {
+  return axios.get(`api/challenges/search/${term}`).then(res => dispatch(getChallenges(res.data)));
+};
+
+export const fetchFilterChallenges = difficulty => dispatch => {
+  return axios
+    .get(`api/challenges/filter/${difficulty}`)
+    .then(res => dispatch(getChallenges(res.data)));
+};
```

Comment:
Maybe think about one function that can take both search term and filters (in an options object maybe?)

`const someFunction = (options) => { // some stuff };`
`const options = { difficulty: 3, searchTerm: "good", author: "kristy", dateCreated: "5/21/2019" }`

Reply:
Thoughts on this?